### DISCUSSION TITLE
Gradio UI launch with "share=True"

### DISCUSSION THREAD

**Discussion by jinwoongyoo** [imartinez/privateGPT#1324] [Q&A]
Hi all,
I am testing privateGPT on a server with GPUs not only local computer. 
Installation went through without any issue and it fires up smoothly with no error.
However, I cannot see the UI through the localhost:8001 due to the fact that I am testing remotely.
I wonder if the privateGPT UI can be set with "share=True" at private_gpt/ui/ui.py as below?

_blocks.launch(debug=False, show_api=False, share=True)

Previously, I had run gradio with the "share=True" option and I was able to open the public url.
With the privateGPT, it does not provide the public url even when I change the private_gpt/ui/ui.py as above.

Any input will be greatly appreciated.
Thanks!

**Comment by annahambi** [comment]
@jinwoongyoo did you manage to expose the user interface from your server?

**Reply by jinwoongyoo** [reply]
@annahambi No, not yet. Do you have any idea?

**Comment by jinwoongyoo** [comment]
No, not yet.

**Reply by MFeras** [reply]
Hello, Did it work with Gradio UI-Start with "share=True or not ?
Thank you very much

**Reply by jinwoongyoo** [reply]
Running Gradio with the "share=True" separately from the privateGPT works. However, within the privateGPT, setting "share=True" option does not generate any public url. Thanks.

**Comment by BradHutchings** [comment]
You need to do the install step with ui and local. I have a recent video which shows the steps for installing on Ubuntu. 

https://youtu.be/elBzQAOpQhE?feature=shared

-Brad

**Comment by jinwoongyoo** [comment]
Thanks for your comment. My original intent was to use it for remote
server, which may not be supported originally. No worries. Thanks. Regards,
Jinwoong

**Comment by LamThienLe** [comment]
Hi,  I am not using Ubuntu. Did anyone make it work on Mac? Using Mac as a host computer for PrivateGPT
Thanks,
